internal/internal_workflow.go

```diff
@@ -924,6 +955,12 @@ func (s *coroutineState) yield(status string) {
 	s.keptBlocked = true
 }
 
+func (s *coroutineState) setState(state ExecutionState) {
+	// TODO(quinn) check blocked
+	s.state = state
+	s.dispatcher.setExecutionState(s.state)
```

Comment:
In other places you set invoke this shared `setExecutionState` you are unsetting it in defer, but here you are just setting it without unsetting. It's a bit hard for me to follow the code to know if this is safe.

Reply:
Yes this is intentional, we sometimes `defer` unsetting because we are only temporally in a state while executing a side effect